libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -184,6 +342,113 @@ class NavigationView : ConstraintLayout, LifecycleObserver {
+                        this@NavigationView.context,
+                        R.drawable.mapbox_navigation_puck_icon
+                    )
+                )
+                setLocationProvider(navigationLocationProvider)
+                enabled = true
+            }
+            MapboxDropInUtils.getLastLocation(this@NavigationView.context) { result ->
+                result.fold(
+                    {
+                        Log.e(TAG, "Error obtaining current location", it)
+                    },
+                    {
+                        it.lastLocation?.apply {
+                            cameraViewModel.consumeLocationUpdate(this)
+                        }
+                    }
+                )
+            }
+        }
+    }
+
+    internal fun addRouteProgressObserver(observer: RouteProgressObserver) {
+        externalRouteProgressObservers.add(observer)
+    }
+
+    internal fun removeRouteProgressObserver(observer: RouteProgressObserver) {
+        externalRouteProgressObservers.remove(observer)
+    }
+
+    internal fun addLocationObserver(observer: LocationObserver) {
+        externalLocationObservers.add(observer)
+    }
+
+    internal fun removeLocationObserver(observer: LocationObserver) {
+        externalLocationObservers.remove(observer)
+    }
+
+    internal fun addRoutesObserver(observer: RoutesObserver) {
+        externalRoutesObservers.add(observer)
+    }
+
+    internal fun removeRoutesObserver(observer: RoutesObserver) {
+        externalRoutesObservers.remove(observer)
+    }
+
+    internal fun addArrivalObserver(observer: ArrivalObserver) {
+        externalArrivalObservers.add(observer)
+    }
+
+    internal fun removeArrivalObserver(observer: ArrivalObserver) {
+        externalArrivalObservers.remove(observer)
+    }
+
+    internal fun addBannerInstructionsObserver(observer: BannerInstructionsObserver) {
+        externalBannerInstructionObservers.add(observer)
+    }
+
+    internal fun removeBannerInstructionsObserver(observer: BannerInstructionsObserver) {
+        externalBannerInstructionObservers.remove(observer)
+    }
+
+    internal fun addTripSessionStateObserver(observer: TripSessionStateObserver) {
+        externalTripSessionStateObservers.add(observer)
+    }
+
+    internal fun removeTripSessionStateObserver(observer: TripSessionStateObserver) {
+        externalTripSessionStateObservers.remove(observer)
+    }
+
+    internal fun getFoobar() {
+        // does nothing
+    }
+
+    private fun getTransitionOptions(
+        locationMatcherResult: LocationMatcherResult
+    ): (ValueAnimator.() -> Unit) {
+        return if (locationMatcherResult.isTeleport) {
+            {
+                duration = 0
+            }
+        } else {
+            {
+                duration = 1000
+            }
+        }
+    }
+
+    fun getMapboxNavigationView(): MapboxNavigationViewApi {
+        return object : MapboxNavigationViewApi {
```

Comment:
Why do we return a new instance on each getter call? We should just cache one instance and make it acecssible.

Reply:
Moved to a single instance.